### ISSUE TITLE
In the REPL some definitions cannot be evaluated if the module doesn't export its dependencies

### ISSUE THREAD

**Issue by lukaszcz** [anoma/juvix#1691]
After loading the module
```
module test001;

open import Stdlib.Prelude;

main : IO;
main := printNatLn (5 + 2 * 3);

end;
```
into the REPL and evaluating `main`, one gets the error:
```
.:1:1: error:
evaluation error: no matching pattern: match 3, 2 with {
  _, zero ↦ zero;
  b, suc a ↦ + b (* a b)
}
```
Everything works as expected if the import is changed to
```
open import Stdlib.Prelude public;
```

It should be possible to evaluate any definition from a loaded module.

**Comment by paulcadman** [comment]
I think that the example in this issue is fixed by https://github.com/anoma/juvix/pull/1775 as the issue described is a symptom of https://github.com/anoma/juvix/issues/1774 - i.e the integer literals in the example are not translated to Nats if builtin `Bool` is not in scope. The workaround of adding `open import Stdlib.Prelude public;` to the module puts builtin `Bool` in scope.

Even though this particular issue is fixed the more general issue of evaluating symbols in the REPL *imported* by the loaded module is not.

For instance if you load the example file and then run: `:t null` in the REPL you get an error because `null` is not exported by the module. The user would expect this to be available in the REPL because the standard library is imported.

**Comment by lukaszcz** [comment]
> I think that the example in this issue is fixed by #1775 as the issue described is a symptom of #1774 - i.e the integer literals in the example are not translated to Nats if builtin `Bool` is not in scope. The workaround of adding `open import Stdlib.Prelude public;` to the module puts builtin `Bool` in scope.
> 

Yes, I think this fixes the issue.

> Even though this particular issue is fixed the more general issue of evaluating symbols in the REPL _imported_ by the loaded module is not.
> 
> For instance if you load the example file and then run: `:t null` in the REPL you get an error because `null` is not exported by the module. The user would expect this to be available in the REPL because the standard library is imported.

This behaviour makes some sense if you think of loading the module as replacing the currently loaded module. We could change the syntax of module loading to be equivalent to "open import". But then in the REPL prompt we should display all loaded modules, not only the last one.

**Comment by lukaszcz** [comment]
Closed by #1775
